lib/features/auth/view/widget/auth_input_field.dart

```diff
@@ -0,0 +1,108 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:telware_cross_platform/core/theme/palette.dart';
+
+class AuthInputField extends ConsumerStatefulWidget {
+  final String name;
+  final double paddingTop;
+  final double paddingBottom;
+  final double paddingLeft;
+  final double paddingRight;
+  final bool isFocused;
+  final bool obscure;
+  final FocusNode focusNode;
+  final FormFieldValidator<String>? validator;
+  final TextEditingController? controller;
+
+  const AuthInputField({
+    super.key,
+    required this.name,
+    this.paddingTop = 0,
+    this.paddingBottom = 0,
+    this.paddingLeft = 0,
+    this.paddingRight = 0,
+    required this.isFocused,
+    required this.focusNode,
+    this.obscure = false,
+    this.validator,
+    this.controller,
+  });
+
+  @override
+  ConsumerState<AuthInputField> createState() => _AuthInputField();
+}
+
+class _AuthInputField extends ConsumerState<AuthInputField> {
+  late bool _obscure;
+  late bool _allowVisibility;
+
+  @override
+  void initState() {
+    super.initState();
+    _allowVisibility = _obscure = widget.obscure;
+  }
+
+  //getters
+  double get paddingTop => widget.paddingTop;
```

Comment:
do we need getters? they are only used once inside this class. and they are used inside this class only